cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -34,6 +34,21 @@ namespace cudf {
 namespace io {
 namespace detail {
 
+namespace {
+class force_init_cuda_context {
+ public:
+  force_init_cuda_context()
+  {
+    // Workaround for https://github.com/rapidsai/cudf/issues/14140, where cuFileDriverOpen errors
+    // out if no CUDA calls have been made before it. This is a no-op if the CUDA context is already
+    // initialized
+    cudaFree(0);
+  }
+};
+
+force_init_cuda_context force_init_cuda_context_instance;
```

Comment:
I believe defining this in global space means that `cudaFree(0)` is technically called before `main()` which is not allowed by the CUDA Developer Guide (see Note at https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#initialization).